storage/reads/response_writer.go

```diff
@@ -91,38 +88,42 @@ func (w *ResponseWriter) WriteResultSet(rs ResultSet) error {
 		"scanned-bytes", fmt.Sprint(stats.ScannedBytes),
 		"scanned-values", fmt.Sprint(stats.ScannedValues)))
 
-	return nil
+	return rs.Err()
 }
 
 func (w *ResponseWriter) WriteGroupResultSet(rs GroupResultSet) error {
 	stats := cursors.CursorStats{}
 	gc := rs.Next()
+	var err error
 	for gc != nil {
 		w.startGroup(gc.Keys(), gc.PartitionKeyVals())
 		for gc.Next() {
+
 			cur := gc.Cursor()
 			if cur == nil {
 				// no data for series key + field combination
 				continue
 			}
 
-			w.startSeries(gc.Tags())
-			w.streamCursor(cur)
+			w.streamCursor(gc.Tags(), cur)
 			if w.err != nil {
 				gc.Close()
 				return w.err
 			}
 			stats.Add(gc.Stats())
 		}
 		gc.Close()
```

Comment:
Check that `gc.Close()` error is reflected in `gc.Err()`

Reply:
From the interface definition:

```
	// Close releases any resources allocated by the GroupCursor.
	Close()

	// Err returns the first error encountered by the GroupCursor.
	Err() error
```

So `Close()` doesn't return an error at all.